src/gear.rs

```diff
@@ -39,6 +39,13 @@ impl Engine for Gear {
     fn digest(&self) -> u32 {
         self.digest.0
     }
+
+    #[inline(always)]
```

Comment:
Let's not do the `always` for this, rolling and digest happen every byte so that's fine, but this is far less frequent. Just `inline` is generally preferable.